### DISCUSSION TITLE
adding gravity with PorousFlowFullySaturated

### DISCUSSION THREAD

**Discussion by asgauthier** [idaholab/moose#24972] [Q&A General]
Hi everyone,
I used PorousFlowFullySaturated in my input file and I tried to add gravity but here is the error i get :
![image](https://github.com/idaholab/moose/assets/60844860/2cdde41d-620e-4c4a-9959-a5e6d8e77a52)
What I'm missing ?
Thanks in advance for your help !

**Comment by GiudGiud** [comment]
Hello

it looks like you should be defining a density material property in all the regions that have gravity.

you should look at examples in the porous flow module to see how they do that 

Guillaume

**Reply by cpgr** [reply]
Yeah, you probably should add a `PorousFlowTotalGravitationalDensityFullySaturatedFromPorosity` material
